src/catala/catala_surface/ast.ml

```diff
@@ -218,7 +218,7 @@ and expression =
 
 type rule = {
   rule_label : ident Pos.marked option;
-  rule_exception_to : ident Pos.marked option;
+  rule_exception_to : ident Pos.marked option option;
```

Comment:
Indeed, we need an `option option` here but we might as well make it a proper sum type with informative variants like

```OCaml
type exception_to = 
  | NotAnException
  | ExceptionToLabel of ident Pos.marked 
  | ExceptionToImplicitBaseCase
```

Reply:
I modified the datatype and the code to use this sum type